packages/MainUI/src/contexts/languageContext.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { createContext, useState, useContext, ReactNode } from 'react';
+
+type Language = 'es' | 'en';
+
+interface LanguageContextType {
+  language: Language;
+  setLanguage: (lang: Language) => void;
+}
+
+const LanguageContext = createContext<LanguageContextType | undefined>(
+  undefined,
+);
+
+export const LanguageProvider: React.FC<{ children: ReactNode }> = ({
+  children,
+}) => {
+  const [language, setLanguage] = useState<Language>('es');
+
+  return (
+    <LanguageContext.Provider value={{ language, setLanguage }}>
+      {children}
+    </LanguageContext.Provider>
+  );
+};
+
+export const useLanguage = () => {
+  const context = useContext(LanguageContext);
+  if (context === undefined) {
```

Comment:
is required === undefined?

Reply:
It is not required but it is not unnecessary either, it prevents runtime errors and makes the code easy to debug if by any chance useLanguage is used outside the LanguageContext. Although if it is certain that we will never use this hook out of the context maybe it could be changed by 

Without verification
  return useContext(LanguageContext) as LanguageContextType;